sdks/python/apache_beam/io/mongodbio.py

```diff
@@ -264,9 +264,13 @@ def display_data(self):
     res['uri'] = _mask_uri_password(self.uri)
     res['database'] = self.db
     res['collection'] = self.coll
-    res['filter'] = json.dumps(self.filter)
+    res['filter'] = json.dumps(
+        self.filter, default=lambda x: 'not_serializable(%s)' % str(x))
     res['projection'] = str(self.projection)
-    res['mongo_client_spec'] = json.dumps(_mask_spec_password(self.spec))
+    res['mongo_client_spec'] = json.dumps(
+        _mask_spec_password(self.spec),
+        default=lambda x: 'not_serializable('
+        '%s)' % str(x))
```

Comment:
one more concern: the spec may have the password (although we try to remove it) - so I wonder if we should avoid calling str on the spec?
Or we leave it up to the user to:

1) Provide a `dict` where the password is protected, or
2) implement a `__str__` method that does not display the password.

Thoughts?

Reply:
I think the spec is probably not that meaningful as display data so I removed it altogether. If user reports that they need to display additional metadata we could then give them an option to pass a dict to the constructor.